units/generate_tasks.go

```diff
@@ -267,7 +267,7 @@ func (j *generateTasksJob) Run(ctx context.Context) {
 
 	defer func() {
 		pErr := recovery.HandlePanicWithError(recover(), nil, fmt.Sprintf("panic in %s job", generateTasksJobName))
-		if err == nil && pErr == nil {
+		if pErr == nil && (err == nil || err.Error() == "") {
```

Comment:
we were getting some errors where err is not a nil interface but has a nil error, alas